subsys/mgmt/ec_host_cmd/ec_host_cmd_handler.c

```diff
@@ -4,26 +4,41 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
-#include <zephyr/drivers/ec_host_cmd_periph/ec_host_cmd_periph.h>
-#include <zephyr/mgmt/ec_host_cmd.h>
+#include <zephyr/mgmt/ec_host_cmd/ec_host_cmd.h>
+#include <zephyr/mgmt/ec_host_cmd/backend.h>
 #include <zephyr/devicetree.h>
 #include <zephyr/kernel.h>
 #include <string.h>
 
-BUILD_ASSERT(DT_HAS_CHOSEN(zephyr_ec_host_interface),
-	"Must choose zephyr,ec-host-interface in device tree");
-
-#define DT_HOST_CMD_DEV DT_CHOSEN(zephyr_ec_host_interface)
-
 #define RX_HEADER_SIZE (sizeof(struct ec_host_cmd_request_header))
 #define TX_HEADER_SIZE (sizeof(struct ec_host_cmd_response_header))
 
-/**
- * Used by host command handlers for their response before going over wire.
- * Host commands handlers will cast this to respective response structures that may have fields of
- * uint32_t or uint64_t, so this buffer must be aligned to protect against the unaligned access.
- */
-static uint8_t tx_buffer[CONFIG_EC_HOST_CMD_HANDLER_TX_BUFFER] __aligned(8);
+#define EC_HOST_CMD_DEFINE(_name)                                                                  \
+	COND_CODE_1(CONFIG_EC_HOST_CMD_HANDLER_RX_BUFFER_DEF,                                      \
+		    (static uint8_t _name##_rx_buffer[CONFIG_EC_HOST_CMD_HANDLER_RX_BUFFER];), ()) \
+	COND_CODE_1(CONFIG_EC_HOST_CMD_HANDLER_TX_BUFFER_DEF,                                      \
+		    (static uint8_t _name##_tx_buffer[CONFIG_EC_HOST_CMD_HANDLER_TX_BUFFER];), ()) \
+	static K_KERNEL_STACK_DEFINE(_name##stack, CONFIG_EC_HOST_CMD_HANDLER_STACK_SIZE);         \
+	static struct k_thread _name##thread;                                                      \
+	static struct ec_host_cmd _name = {                                                        \
+		.rx_ctx =                                                                          \
+			{                                                                          \
+				.buf = COND_CODE_1(CONFIG_EC_HOST_CMD_HANDLER_RX_BUFFER_DEF,       \
+						   (_name##_rx_buffer), (NULL)),                   \
+			},                                                                         \
+		.tx =                                                                              \
+			{                                                                          \
+				.buf = COND_CODE_1(CONFIG_EC_HOST_CMD_HANDLER_TX_BUFFER_DEF,       \
+						   (_name##_tx_buffer), (NULL)),                   \
+				.len_max =                                                         \
+					COND_CODE_1(CONFIG_EC_HOST_CMD_HANDLER_TX_BUFFER_DEF,      \
```

Comment:
If you allow to use multiple backends, how do you define the buffer size? Multiple backends can have different requirements for the size, should the user just know that this buffer must be big enough for every backend?

Reply:
Every backend defines default buffers sizes. If multiple backends are use, user has to set the config to meet requirements of the largest one. I updated the help message.